pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -382,6 +385,24 @@ func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *crdv1alpha1.C
 	return internalNetworkPolicy
 }
 
+// translateServiceAccountInPeer translates a crdv1alpha1.NetworkPolicyPeer. If the
+// original NetworkPolicyPeer has ServiceAccount, it will be translated to a
+// NetworkPolicyPeer with PodSelector+NamespaceSelector. Otherwise, the original
+// NetworkPolicyPeer will be returned.
+func (n *NetworkPolicyController) translateServiceAccountInPeer(originalPeer crdv1alpha1.NetworkPolicyPeer) crdv1alpha1.NetworkPolicyPeer {
+	if originalPeer.ServiceAccount == nil {
+		return originalPeer
+	}
+	return crdv1alpha1.NetworkPolicyPeer{
+		PodSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{grouping.CustomLabelKeyPrefix + grouping.ServiceAccountLabelKey: originalPeer.ServiceAccount.Name},
+		},
+		NamespaceSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"kubernetes.io/metadata.name": originalPeer.ServiceAccount.Namespace},
```

Comment:
Translating a service account name selector to namespaceSelector + PodSelector may have two problems:
1. It doesn't work for K8s versions that don't add default namespace label.
2. NamespaceSelector has higher cost than Namespace name. Although it will match one namespace eventually, the matching process doesn't know it and have to scan all namespaces.

Maybe we use if else when creating group:
```
if at.ServiceAccount != nil {
	atg = n.createAppliedToGroup(at.ServiceAccount.Namespace, serviceAccountNameSelector(at.ServiceAccount.Name), nil, nil)
} else {
	atg = n.createAppliedToGroup(ns, translatedAT.PodSelector, nil, translatedAT.ExternalEntitySelector)
}
```

Reply:
Good idea. Changed.